tests/test_scripts.py

```diff
@@ -561,6 +563,104 @@ def test_workspace_digest(tmpdir, script_runner, algorithms, do_json):
         }
 
 
+@pytest.mark.parametrize(
+    "archive",
+    [
+        "https://www.hepdata.net/record/resource/1408476?view=true",
+        "https://doi.org/10.17182/hepdata.89408.v1/r2",
+    ],
+)
+def test_patchset_download(datadir, script_runner, archive):
+    command = f'pyhf contrib download {archive} {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert ret.success
+
+    # Run with all optional flags
+    command = f'pyhf contrib download --verbose --force {archive} {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert ret.success
+
+    command = f'pyhf contrib download --verbose https://www.fail.org/record/resource/1234567 {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert not ret.success
+    assert (
+        "pyhf.exceptions.InvalidArchiveHost: www.fail.org is not an approved archive host"
+        in ret.stderr
+    )
+    command = f'pyhf contrib download --verbose --force https://www.fail.org/record/resource/1234567 {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert not ret.success
+    # Python 3.6 has different return error than 3.7, 3.8
+    assert (
+        "ssl.CertificateError: hostname 'www.fail.org' doesn't match"
+        or "certificate verify failed: Hostname mismatch, certificate is not valid for 'www.fail.org'."
+        in ret.stderr
+    )
+
+
+def test_missing_contrib_extra(capsys):
+    sys.modules["requests"] = None
+    # Force clean state
+    _del_modules = [
+        "pyhf.cli",
+        "pyhf.cli.contrib",
+        "pyhf.contrib",
+        "pyhf.contrib.utils",
+    ]
+    for module in _del_modules:
+        if module in sys.modules.keys():
+            del sys.modules[module]
```

Comment:
This was found by trial and error. While at the moment it works, it is pretty ugly and not very satisfying. Critique here is most welcome.

Reply:
I should probably use the `mocker` fixture here with

```python
    mocker.patch.dict(sys.modules, {"requests": None})
```
and then also mock the `del` of the modules, but I'm not sure how to get `del` working with `mocker.patch.dict`.

Reply:
Following this https://youtu.be/PXu3KCMT3l4 I decided to just use `mock`